### ISSUE TITLE
Float value labels instead of strings

### ISSUE THREAD

**Issue by extremus** [robsmith100/sav#4]
Variables of type *string* have *float* values in `meta.valueLabels`:

``` json
{
    "appliesToShortNames": [ "G" ],
    "entries": [
        {
           "val": 6.013470016999143e-154,
           "label": "First"
        },
        {
           "val": 6.013470016999144e-154,
           "label": "Second"
        }
    ]
}
```

Here is what I expected:

``` json
{
    "appliesToShortNames": [ "G" ],
    "entries": [
        {
           "val": "X",
           "label": "First"
        },
        {
           "val": "Y",
           "label": "Second"
        }
    ]
}
```

![shot-220506-155152](https://user-images.githubusercontent.com/1205348/167134930-00ff63f8-a886-4b94-a5bd-e1f7ed412c5a.png)

- - - -

* macOS v12.3.1
* node v14.15.4
* sav-reader v2.0.2

**Comment by robsmith100** [comment]
Thanks! I'm looking into this. I didn't realize string variables could have value labels, but now I see they do. I appreciate this issue being raised.

**Comment by robsmith100** [comment]
@extremus I see PSPPIRE in your screenshot. Did you write the file using PSPPIRE? If not, what utility wrote the .sav file?

**Comment by extremus** [comment]
> I see PSPPIRE in your screenshot. Did you write the file using PSPPIRE? If not, what utility wrote the .sav file?

Yes, it was *PSPP*. But, I found that issue on another file, that was produced via *IBM SPSS*:  `IBM SPSS STATISTICS 64-bit MS Windows 25.0.0.0`

**Comment by robsmith100** [comment]
I believe I understand the issue now. From observation it appears that if the variable Type is String with Width <= 8 and value labels exist, a ValueLabelRecord will be written for the variable. For each entry in the ValueLabelRecord, this reader utility should interpret the value as a string whereas now it interprets the 8 bytes as a Double. Also I observed that if the variable Type is String and its Width is > 8, value labels are stored differently. Instead, a new InfoRecord type is written (subtype=21) with the valuelabels. I'll make a patch ASAP. As a workaround for string vars of width <= 8, one could convert the entry's "val" into 8 bytes and use String.fromCharCode on each byte.

**Comment by extremus** [comment]
Thanks! 
I do have some string variables of width 9 and more in my .sav files. Looks like it also explains why there is no value lables on such varibles.

**Comment by robsmith100** [comment]
I created PR #5 for this, but I'm going to test it for a day or two before merging.

**Comment by robsmith100** [comment]
Closing due to merged PR #5
